x-pack/plugins/apm/server/routes/fleet/api_keys/create_apm_api_keys.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { CoreStart, Logger } from '@kbn/core/server';
+
+export async function createApmSourceMapApiKey({
+  coreStart,
+  logger,
+}: {
+  coreStart: CoreStart;
+  logger: Logger;
+}) {
+  logger.debug('Creating source map API Key');
+
+  const response =
+    await coreStart.elasticsearch.client.asInternalUser.security.createApiKey({
+      body: {
+        name: 'apm-source-map-index-api-key',
+        metadata: {
+          description:
+            'Created for APM Server to be able to retrieve source maps directly from Elasticsearch',
+          application: 'apm',
+          consumer: 'apm-server',
+        },
+        role_descriptors: {
+          apmSystemIndices: {
+            cluster: ['all'],
+            index: [
+              {
+                names: ['.apm-source-map'],
+                privileges: ['all'],
```

Comment:
@kruskall I'm changing this to `read`. Ok?

Reply:
Yep, I think this is fine we only search sourcemaps